src/sources/kubernetes_logs/pod_metadata_annotator.rs

```diff
@@ -153,112 +170,235 @@ fn annotate_from_file_info(
-        if let (Some(key), Some(val)) = (&key.path, val) {
-            log.insert(key, val.to_owned());
+        if let Some(value) = value {
+            let legacy_key = legacy_key
+                .path
+                .as_ref()
+                .map(|k| &k.path)
+                .map(LegacyKey::Overwrite);
+
+            log_namespace.insert_source_metadata(
+                Config::NAME,
+                log,
+                legacy_key,
+                *metadata_key,
+                value.to_owned(),
+            );
         }
     }
 
-    if let (Some(key), Some(owner_references)) =
-        (&fields_spec.pod_owner.path, &metadata.owner_references)
-    {
-        log.insert(
-            key,
+    if let Some(owner_references) = &metadata.owner_references {
+        let legacy_key = fields_spec
+            .pod_owner
+            .path
+            .as_ref()
+            .map(|k| &k.path)
+            .map(LegacyKey::Overwrite);
+
+        log_namespace.insert_source_metadata(
+            Config::NAME,
+            log,
+            legacy_key,
+            path!("pod_owner"),
             format!("{}/{}", owner_references[0].kind, owner_references[0].name),
-        );
+        )
     }
 
     if let Some(labels) = &metadata.labels {
-        if let Some(pod_label_prefix) = &fields_spec.pod_labels.path {
-            for (key, val) in labels.iter() {
-                let key_path = path!(key);
-                log.insert(
-                    (PathPrefix::Event, (&pod_label_prefix.path).concat(key_path)),
-                    val.to_owned(),
-                );
-            }
+        for (key, value) in labels.iter() {
+            let key_path = path!(key);
+            let legacy_key = fields_spec
+                .pod_labels
+                .path
+                .as_ref()
+                .map(|k| &k.path)
+                .map(|k| k.concat(key_path))
+                .map(LegacyKey::Overwrite);
+
+            log_namespace.insert_source_metadata(
+                Config::NAME,
+                log,
+                legacy_key,
+                path!("pod_labels", key),
+                value.to_owned(),
+            )
         }
     }
 
     if let Some(annotations) = &metadata.annotations {
-        if let Some(pod_annotations_prefix) = &fields_spec.pod_annotations.path {
-            for (key, val) in annotations.iter() {
-                let key_path = path!(key);
-                log.insert(
-                    (
-                        PathPrefix::Event,
-                        (&pod_annotations_prefix.path).concat(key_path),
-                    ),
-                    val.to_owned(),
-                );
-            }
+        for (key, value) in annotations.iter() {
+            let key_path = path!(key);
+            let legacy_key = fields_spec
+                .pod_annotations
+                .path
+                .as_ref()
+                .map(|k| &k.path)
+                .map(|k| k.concat(key_path))
```

Comment:
same here

Reply:
Resolved in 3161817cea1c1d85fd5d1050a877208c53db6bb2